### PR TITLE
Add GitHub configuration

### PR DESCRIPTION
**Pull request by slushpupie**
* Issue templates
  * Pull Request template
  * Stale issue workflow
  * CodeQL workflow
  * Workflows to build a Docker image from a PR, and clean it up
  * Update release/test workflows to checkout@v4

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/03_documentation.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  
+### The page URLs
+  
+(List the URLs to update here)
+
+* (examples in this repository)
+
+### Requirements
+
+Please make sure if this topic is specific to this SDK. For general questions/issues about Slack API platform or its server-side, could you submit questions at https://my.slack.com/help/requests/new instead. :bow:
```
Comment (MattKempfert) [inline .github/ISSUE_TEMPLATE/03_documentation.md:19]: Can we reword this a bit?

~Please make sure if this topic is specific to this SDK.~
Please make sure this is a documentation issue related to Emoji Manager.

File: .github/ISSUE_TEMPLATE/04_bug.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  
+(Describe what you expected to happen)
+  
+#### Actual result:
+  
+(Describe what actually happened with logs, screenshots)
+
+### Requirements
+
+Please read the [Contributing guidelines](https://github.com/slackapi/java-slack-sdk/blob/main/.github/contributing.md) and [Code of Conduct](https://slackhq.github.io/code-of-conduct) before creating this issue or pull request. By submitting, you agree to those rules.
```
Comment (MattKempfert) [inline .github/ISSUE_TEMPLATE/04_bug.md:46]: Are these supposed to be links to Slack's SDK contribute guideline and code of conduct?

File: .github/pull_request_template.md [hunk 3]
```diff
@@ -0,0 +1,7 @@
+(Describe the goal of this PR. Mention any related Issue numbers)
+
+## Requirements
+
+You will need to sign the CLA if this is your first time contributing.
+
+Please read the [Contributing guidelines](https://github.com/slackapi/java-slack-sdk/blob/main/.github/contributing.md) and [Code of Conduct](https://slackhq.github.io/code-of-conduct) before creating this issue or pull request. By submitting, you agree to those rules.
```
Comment (MattKempfert) [inline .github/pull_request_template.md:7]: Are these supposed to be links to Slack's SDK contribute guideline and code of conduct?

File: .github/workflows/pr-merge.yml [hunk 4]
```diff
@@ -0,0 +1,16 @@
+name: PR Image Docker Cleanup
+on:
+  pull_request:
+    types: [closed]
```
Comment (dtanner) [inline .github/workflows/pr-merge.yml:4]: does this need to be further filtered to only those issues that had an image built as a result of the `build-docker-image` workflow?

**Review by dtanner — APPROVED** [review]
question about potentially trying to delete images that might not exist, but otherwise 👍

File: .github/ISSUE_TEMPLATE/04_bug.md (hunk 2 above)
Comment (slushpupie) [reply .github/ISSUE_TEMPLATE/04_bug.md:46]: Doh! I borrowed from Slack and missed a few..  will update.

File: .github/workflows/pr-merge.yml (hunk 4 above)
Comment (slushpupie) [reply .github/workflows/pr-merge.yml:4]: Ideally yes..  but how?  Also, there isn't much downside to running this on PRs that didn't generate an image.

**Review by MattKempfert — APPROVED** [review]

**Pull request merged**